subworkflows/local/prepare_genome.nf

```diff
@@ -0,0 +1,91 @@
+
+include { BOWTIE_BUILD        } from '../../modules/nf-core/bowtie/build/main'
+include { BOWTIE2_BUILD       } from '../../modules/nf-core/bowtie2/build/main'
+include { BWA_INDEX           } from '../../modules/nf-core/bwa/index/main'
+include { HISAT2_EXTRACTSPLICESITES } from '../../modules/nf-core/hisat2/extractsplicesites/main'
+include { HISAT2_BUILD        } from '../../modules/nf-core/hisat2/build/main'
+include { STAR_GENOMEGENERATE } from '../../modules/nf-core/star/genomegenerate/main'
+include { SEGEMEHL_INDEX      } from '../../modules/nf-core/segemehl/index/main'
+
+workflow PREPARE_GENOME {
+
+    take:
+    fasta
+    gtf
+
+    main:
+    ch_versions = Channel.empty()
+
+    ch_fasta = Channel.fromPath(fasta)
+
+    // MapSplice & find_circ requires reference genome to be split per chromosome:
+    if( ( params.tool.contains('mapsplice') || params.tool.contains('find_circ') ) && params.module.contains('circrna_discovery') ){
+        file("${params.outdir}/genome/chromosomes").mkdirs()
+        ch_fasta.splitFasta( record: [id:true] )
+                .map{ record -> record.id.toString() }
+                .set{ ID }
+
+        ch_fasta.splitFasta( file: true )
+                .merge( ID ).map{ file, id -> file.copyTo("${params.outdir}/genome/chromosomes/${id}.fa") }
+
+    stage_chromosomes = Channel.value("${workflow.launchDir}/${params.outdir}/genome/chromosomes")
+    }
+
+    // some index procs use tuple, some dont -_-
```

Comment:
You can also `nf-core modules patch` these out if it's easier for you ;) but we are trying to move so everything has `meta`